server/src/api/resources/trip/trip.controller.js

```diff
@@ -137,3 +144,37 @@ export const repeatTrip = (req, res) => {
   delete updatedRequest.body.id
   createTrip(updatedRequest, res)
 }
+
+export const uploadPics = (req, res) => {
+  const id = req.params.id
+  cloudinary.v2.uploader.upload(req.body.image, (err, result) => {
+    Trip.findOneAndUpdate(
+      { _id: id },
+      { $push: { tripPics: result.url } },
+      { returnOriginal: false }
+    ).then(oldTrip => {
+      Trip.findOne({ _id: oldTrip.id })
+        .then(newTrip => {
+          res.status(200).json(newTrip)
+        })
+        .catch(() => {
+          res.status(404).json("Not Found")
+        })
+        .catch(error => {
```

Comment:
This `.catch` block should be chained _after_ `Trip.findOneAndUpdate`. Currently, you have two `.catch` blocks chained after `Trip.findOne`

Reply:
ok yeah I deleted the second catch. Thanks for the catch